### PR TITLE
Updated Readme to reflect current features

### PR DESCRIPTION
**Pull request by TheOtherBrian1**
## What kind of change does this PR introduce?

The documentation is dated. Some sections have not been updated in over two years. Recognizing the importance of robust and current documentation, I expanded the Readme to be comprehensive, covering all aspects of the project, including newly introduced features, such as the net.http_delete function, practical examples to guide users, and clear instructions for installation.

## What is the current behavior?

Prior to this update, the Readme document provided minimal guidance. It lacked a reference to important functions such as net.http_delete and did not offer users practical examples or technical explanations.

## What is the new behavior?

The revised Readme now serves as a comprehensive guide to the PG_NET extension. It includes an overview of the project, a detailed table of contents for quick navigation, and a variety of examples demonstrating the usage of the extension.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,606 @@
...
+    </a>
+    <a href="https://github.com/supabase/pg_net/actions">
+        <img src="https://github.com/supabase/pg_net/actions/workflows/main.yml/badge.svg" alt="Tests" height="18">
+    </a>
+</p>
+
+<strong>Documentation</strong>:<a href="https://supabase.github.io/pg_net" target="_blank">https://supabase.github.io/pg_net</a>
+
+<hr />
+<h1>Contents</h1>
```
Comment (steve-chavez) [inline README.md:20]: @TheOtherBrian1 Awesome PR!

Any reason on why not use markdown headings here? e.g.

```suggestion
# Contents
```

Comment (steve-chavez) [reply README.md:20]: Seems we could avoid some HTML tags in other part too.

Comment (TheOtherBrian1) [reply README.md:20]: @steve-chavez When I began editing the file, I noticed that the majority of the document was written in HTML. For consistency purposes, I decided to prioritize HTML over markdown. If it is preferable, I could change it all to Markdown.

Comment (steve-chavez) [reply README.md:20]: >  If it is preferable, I could change it all to Markdown.

That would be great. Thanks!

Comment (TheOtherBrian1) [reply README.md]: @steve-chavez It's all in markdown now

**Review by steve-chavez — APPROVED** [review]

**Pull request merged**